airflow/models/dagbag.py

```diff
@@ -381,34 +381,30 @@ def _load_modules_from_zip(self, filepath, safe_mode):
     def _process_modules(self, filepath, mods, file_last_changed_on_disk):
         from airflow.models.dag import DAG  # Avoid circular import
 
-        is_zipfile = zipfile.is_zipfile(filepath)
-        top_level_dags = [o for m in mods for o in list(m.__dict__.values()) if isinstance(o, DAG)]
+        top_level_dags = ((o, m) for m in mods for o in list(m.__dict__.values()) if isinstance(o, DAG))
```

Comment:
`dict.values()` actually implements `__contains__` so the extra `list()` call is not needed